test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs

```diff
@@ -0,0 +1,161 @@
+{
+    [TestFixture]
+    class DynamoAnalyticsClientTest : DynamoModelTestBase
+    {
+        private const string DEFAULT_RETURN_VALUE = "Default Not Found";
+        private const string SHUTDOWN_TYPE_NAME = "CleanShutdown";
+        private const string REG_KEY = "HKEY_CURRENT_USER\\Software\\DynamoStability";
+        private const string ASSUMING_CRASHING_SHUTDOWN_VALUE = "assumingCrashing";
+
+        //We need to override this function because the one in DynamoModelTestBase is setting StartInTestMode = true
+        protected override DynamoModel.IStartConfiguration CreateStartConfiguration(IPreferences settings)
+        {
+            var preferenceSettings = new PreferenceSettings
+            {
+                IsUsageReportingApproved = true //This will enable to execute the Get method from DynamoAnalyticsClient.Logger
+            };
+
+            return new DynamoModel.DefaultStartConfiguration()
+            {
+                PathResolver = pathResolver,
+                StartInTestMode = false,
+                GeometryFactoryPath = preloader.GeometryFactoryPath,
+                Preferences = preferenceSettings,
+                ProcessMode = TaskProcessMode.Synchronous
+            };
+        }
+
+        [SetUp]
+        public void Init()
+        {
+
+        }
+
+        /// <summary>
+        /// This test method will execute the StabilityCookie.WriteCrashingShutdown(); method
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void TestDynamoAnalyticsSession_Dispose()
+        {
+            //Simulating that the system crashed by some reason
+            DynamoModel.IsCrashing = true; //This will allow to execute the  StabilityCookie.WriteCrashingShutdown(); method inside the Dispose method
+
+            //Arrange
+            // Open/Run XML test graph
+            string openPath = Path.Combine(TestDirectory, @"core\Angle.dyn");
+            RunModel(openPath);
+            int InitialNodesCount = CurrentDynamoModel.CurrentWorkspace.Nodes.Count();
+
+            // Convert a DSFunction node Line.ByPointDirectionLength to custom node.
+            var workspace = CurrentDynamoModel.CurrentWorkspace;
+            var node = workspace.Nodes.OfType<DSFunction>().First();
+
+            List<NodeModel> selectionSet = new List<NodeModel>() { node };
+            var customWorkspace = CurrentDynamoModel.CustomNodeManager.Collapse(
+                selectionSet.AsEnumerable(),
+                Enumerable.Empty<Dynamo.Graph.Notes.NoteModel>(),
+                CurrentDynamoModel.CurrentWorkspace,
+                true,
+                new FunctionNamePromptEventArgs
+                {
+                    Category = "Testing",
+                    Description = "",
+                    Name = "__AnalyticsServiceTest__",
+                    Success = true
+                }) as CustomNodeWorkspaceModel;
+
+            //Act
+            //This will execute the 
```

Comment:
incomplete comment

Reply:
@QilongTang 
I modified the cs file and completed the comment and also added more description
see:
https://github.com/DynamoDS/Dynamo/pull/10546/commits/a79f52d69895469c578f472e757ebae264e1d32b